codespeed/static/js/timeline.js

```diff
@@ -158,8 +158,18 @@ function renderPlot(data) {
     }
     //determine significant digits
     var digits = 2;
-    var value = Math.min.apply( Math, lastvalues );
-    if (value !== 0) {
+
+    // determine smallest non-negative value in lastvalues
+    // since missing values can be represented as -1, we need to be a bit smarter than Math.min
```

Comment:
Since `lastvalues` is only used for determining significant digits, we might as well keep the more readable and efficient `Math.min` and instead push to `lastvalues` (L157) already cleansed values, that is, only positive integers or zero:
```javascript
lastvalues.push(Math.max(0, data.branches[branch][exe_id][0][1]));
```

Or even we could just store the min value inline, while we loop, and not declare a `lastvalues` array at all.